.vscode/settings.json

```diff
@@ -1,4 +1,10 @@
 {
-  "typescript.tsdk": "node_modules/typescript/lib",
-  "typescript.enablePromptUseWorkspaceTsdk": true
-}
\ No newline at end of file
+    "typescript.tsdk": "node_modules/typescript/lib",
+    "typescript.enablePromptUseWorkspaceTsdk": true,
+    "editor.tabSize": 2,
```

Comment:
I see `tabSize` is set to `2` in two different places, but the files are still formatted with 4 spaces.
My guess is that you need to set `"tabWidth": 2` in `.prettierrc.json`. If that solves it, I would proceed to remove the other settings (if it doesn't break anything), as having unnecessary stuff in the codebase tends to proliferate and create confusion.

Reply:
Actually, I just noticed there's a `prettier` section in `package.json` where `tabWidth` is set to `4`.
I suggest merging all `prettier` settings into a single place (my personal preference would be the `.prettierrc.json` file)

Reply:
I removed the prettier section from `package.json`. Also removed `editor.tabSize` from `.vscode/settings`. I presume that means that contributors have to have at least `prettier` or `editorconfig` extensions enabled in order for the formatting to work.

Also formatted all the codebase once again.

Thanks for you eagle eyes @lucian-v